app/components/dds/dds-project-files-picker.js

```diff
@@ -1,10 +1,20 @@
 import Ember from 'ember';
+import FilenameFilter from 'bespin-ui/utils/filename-filter';
 
 const DDSProjectFilesPicker = Ember.Component.extend({
   projects: null,
   project: null,
   selectedResources: null,
-  onFilePicked: function(/* file */) {},
+  onFilePicked: (/* file */) => {},
+  fileFilter: (/*item, index, enumerable*/) =>  {return true},
+  init() {
+    this._super(...arguments);
+    const fileNameRegexStr = this.get('fileNameRegexStr');
+    if (fileNameRegexStr) {
+      const fileFilter = FilenameFilter.create({ fileNameRegexStr: fileNameRegexStr});
+      this.set('fileFilter', fileFilter.filter.bind(fileFilter));
```

Comment:
It's not clear from the interface what the preferred way provide a filter is. Should we provide a fileFilter function, a fileNameRegexStr, or both?

It might be more clear for the `fileFilter` property to be computed based on the `fileNameRegexStr` instead of computing it once in the initializer.